pkg/operator/credentialsrequest/actuator/actuator.go

```diff
@@ -32,6 +36,12 @@ type Actuator interface {
 	Update(context.Context, *minterv1.CredentialsRequest) error
 	// Checks if the credentials currently exists.
 	Exists(context.Context, *minterv1.CredentialsRequest) (bool, error)
+	// Upgradeable returns a ClusterOperator Upgradeable condition to indicate whether or not this cluster can
+	// be safely upgraded to the next "minor" (4.y) Openshift release.
+	Upgradeable(operatorv1.CloudCredentialsMode) *configv1.ClusterOperatorStatusCondition
+	// GetUpcomingCredSecrets returns a slice of NamespacedNames for secrets holding credentials that we know are coming in the next OpenShift release.
+	// Used to pre-check if Upgradeable condition should be true or false for manual mode deployments of the credentials operator.
+	GetUpcomingCredSecrets() []types.NamespacedName
```

Comment:
Is this function temporary? How are we planning to maintain this function for future releases? For example 4.6 -> 4.7 will have different set of upcoming secrets. Does it makes sense to have something like  `GetUpcomingCredSecretsForRelease(release string)`

Reply:
My thought was that we do not need to maintain past releases, once we're on 4.7, there's no need to keep checking for the secrets that were introduced in 4.7 because those are now handled by their respective CredentialsRequests. We now want to transition to 4.8, so I was thinking we just need one release at a time in here, whatever is coming next. The function should be updated every release but wouldn't cause major problems if we forgot because the secrets should exist by that point.